core/src/main/kotlin/pages/RendererSpecificPage.kt

```diff
@@ -1,8 +1,12 @@
 package org.jetbrains.dokka.pages
 
+import org.jetbrains.dokka.links.DRI
+import org.jetbrains.dokka.model.DisplaySourceSet
 import org.jetbrains.dokka.renderers.Renderer
 import kotlin.reflect.KClass
 
+typealias LocationResolver = (DRI, Set<DisplaySourceSet>) -> String
```

Comment:
I really like this idea. If you want you can even change it from `typealias` to `fun interface`.